src/Sylius/Bundle/ShopBundle/Resources/config/app/config.yml

```diff
@@ -54,3 +54,20 @@ liip_imagine:
 
 sonata_block:
     default_contexts: ~
+
+sylius_ui:
+    events:
+        sylius.shop.layout.topbar:
+            blocks:
+                before_switchers_legacy:
+                    template: "@SyliusShop/Block/Layout/_beforeSwitchersLegacy.html.twig"
+                currency_switcher:
+                    template: "@SyliusShop/Block/Layout/_currencySwitcher.html.twig"
+                locale_switcher:
+                    template: "@SyliusShop/Block/Layout/_localeSwitcher.html.twig"
```

Comment:
Need to figure out priorities for those blocks. Any ideas?

Reply:
I used the following conventions:
 - last meaningful block starts at priority: 10
 - other meaningful blocks are having a bigger priority than the last one by +10
 - legacy Sonata events are +5 / -5 (before / after) from the related meaningful block